lib/rucio/common/config.py

```diff
@@ -76,9 +76,23 @@ def config_get_float(section, option):
     return __CONFIG.getfloat(section, option)
 
 
-def config_get_bool(section, option):
-    """Return the boolean value for a given option in a section"""
-    return __CONFIG.getboolean(section, option)
+def config_get_bool(section, option, raise_exception=True, default=None):
```

Comment:
i don't get the reason for raise_exception, that should be covered by the default?

Reply:
no, it's not covered by the default as the default value is only considered if raise_exception is set and None can be a possible default value. raise_exception has been also introduced to be backward compatible.

Reply:
no, it's not covered by the default as the default value is only considered if raise_exception is set and None can be a possible default value. raise_exception has been also introduced to be backward compatible.

Reply:
my browser crashed, somehow my review broke ;-)

ok.

just one thing, we need to check that the default is of the actual datatype (or can be converted), and otherwise throw a TypeError like .getboolean(), .getfloat(), etc. would. otherwise by mistake one could continue with the wrong datatype.

Reply:
just added bool conversion if not None to be protected in that case.

Reply:
thx.

do we want to extend get_int and get_float in the same way?

Reply:
maybe later, it's not anymore a blocker for readthedocs now